### PR TITLE
[receiver/redis] Apply new mdatagen metrics builder

### PR DESCRIPTION
**Pull request by dmitryax**
This change migrates redis receiver to use new mdatagen metrics builder. 

There are no functional changes on the metrics output except for:
 - Metrics reordering.
 - Squashing data points from one metric that were reported under several duplicated metric objects.

This change adds ability to disable collection of particular metrics: to be documented later.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/10904

### CODE REVIEW

File: cmd/mdatagen/metrics_v2.tmpl [hunk 1]
```diff
@@ -150,6 +150,19 @@ func (mb *MetricsBuilder) Record{{ $name.Render }}DataPoint(ts pdata.Timestamp
 }
 {{ end }}
 
+// Reset resets metrics builder to its initial state.
+func (mb *MetricsBuilder) Reset(options ...metricBuilderOption) {
+	mb.startTime = pdata.NewTimestampFromTime(time.Now())
+	for _, op := range options {
+		op(mb)
+	}
+}
```
Comment (bogdandrutu) [inline cmd/mdatagen/metrics_v2.tmpl:159]: Why not replacing the *MetricsBuilder entirely by calling the constructor?

Comment (dmitryax) [reply cmd/mdatagen/metrics_v2.tmpl:159]: We can. It's just couple more things the redis scraper needs to pass around (as part of the scraper struct) that are already part of the builder: startTime and user settings. But I think it's ok for this case. I'll update it

Comment (bogdandrutu) [reply cmd/mdatagen/metrics_v2.tmpl:159]: May be worth documenting when a "reset" is needed, is it not the case that starttime changed?

Comment (dmitryax) [reply cmd/mdatagen/metrics_v2.tmpl:159]: Yes, changed (or reset) startTime is the only reason I can think of.

Comment (dmitryax) [reply cmd/mdatagen/metrics_v2.tmpl]: Updated the doc string. Still not sure if we need this method or ask scraper authors to recreate the builder instead. WDYT?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**